src/promptflow-tools/requirements.txt

```diff
@@ -1,2 +1,3 @@
 google-search-results==2.4.1
 promptflow
+azure-ai-ml
```

Comment:
pf-tools should have not dependency on azure by default. please remove change in requirements.txt.

Reply:
@chjinche , we need this to refer to MLClient and pull the online endpoints to show in the dropdown. If I remove it from here then the test will start to fail.

![image](https://github.com/microsoft/promptflow/assets/75808410/da442026-d749-4e09-bda5-dce742892608)

Reply:
The common practice is to add 'azure' related dependency as an `extra_requires`. You may see this as an reference https://github.com/microsoft/promptflow/blob/ef22b0f8727623a6190575c1770270fa04c891f7/examples/tools/tool-package-quickstart/setup.py#L16

Reply:
as for tool test issue, you may add the dependency you need to tool_tests, like this https://github.com/microsoft/promptflow/blob/ef22b0f8727623a6190575c1770270fa04c891f7/.github/workflows/tools_tests.yml#L27

Reply:
> as for tool test issue, you may add the dependency you need to tool_tests, like this
> 
> https://github.com/microsoft/promptflow/blob/ef22b0f8727623a6190575c1770270fa04c891f7/.github/workflows/tools_tests.yml#L27

Does this change needs to go out in a separate PR? Seems including this change in this PR has no effect on how the CI is setup.

Reply:
@crazygao , help clarify that this was due to merge conflicts on the PR. Resolving those got the package reflected in CI test setup.